bootstrap/bootstrap_test.go

```diff
@@ -0,0 +1,88 @@
+package bootstrap
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestRetryInterceptor(t *testing.T) {
+	expecterErr := status.Error(codes.ResourceExhausted, "resource exhausted")
+	interceptor := retryInterceptor(100*time.Millisecond, 10*time.Millisecond)
+
+	testCases := []struct {
+		name           string
+		invoker        func(callCount int) error
+		maxRequestTime time.Duration
+		callCount      int // expect exact count
+		minCallCount   int // min, for when using a timeout
+		expectedErr    error
+	}{
+		{
+			name: "no error",
+			invoker: func(callCount int) error {
+				return nil
+			},
+			maxRequestTime: 10 * time.Millisecond,
+			callCount:      1,
+			expectedErr:    nil,
+		},
+		{
+			name: "succeeds on 3rd attempt",
+			invoker: func(callCount int) error {
+				if callCount >= 3 {
+					return nil
+				}
+				return expecterErr
+			},
+			maxRequestTime: 40 * time.Millisecond,
+			callCount:      3,
+			expectedErr:    nil,
+		},
+		{
+			name: "fails after timeout",
+			invoker: func(callCount int) error {
+				return expecterErr
+			},
+			maxRequestTime: 150 * time.Millisecond, // add a buffer for test slowness
+			minCallCount:   10,
+			expectedErr:    expecterErr,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
```

Comment:
not including this was causing lint errors which is strange since it should be OK since go1.22.

Reply:
This is due to the `golangci-lint` version that we use: https://github.com/onflow/flow-evm-gateway/blob/main/.github/workflows/ci.yml#L41. I tried with the latest (`golangci-lint@v1.63.4`), and it doesn't produce this lint error. I will open up a new PR to update it.

Reply:
Opened https://github.com/onflow/flow-evm-gateway/pull/718 .